test/tasks/ods-build-go_test.go

```diff
@@ -64,6 +64,60 @@ func TestTaskODSBuildGo(t *testing.T) {
 					}
 				},
 			},
+			"build go app with build caching": {
+				WorkspaceDirMapping: map[string]string{"source": "go-sample-app"},
+				PreRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+					ctxt.ODS = tasktesting.SetupGitRepo(t, ctxt.Namespace, wsDir)
+					ctxt.Params = map[string]string{
+						"go-os":              runtime.GOOS,
+						"go-arch":            runtime.GOARCH,
+						"sonar-quality-gate": "true",
+						"cache-build":        "true",
+					}
+				},
+				WantRunSuccess: true,
+				PostRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+
+					checkFilesExist(t, wsDir,
+						"docker/Dockerfile",
+						"docker/app",
+						filepath.Join(pipelinectxt.LintReportsPath, "report.txt"),
+						filepath.Join(pipelinectxt.XUnitReportsPath, "report.xml"),
+						filepath.Join(pipelinectxt.CodeCoveragesPath, "coverage.out"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "analysis-report.md"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "issues-report.csv"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "quality-gate.json"),
+					)
+
+					sonarProject := sonar.ProjectKey(ctxt.ODS, "")
+					checkSonarQualityGate(t, ctxt.Clients.KubernetesClientSet, ctxt.Namespace, sonarProject, true, "OK")
+
+					// This is not available when build skipping as the default is
+					// supplied on the second repeat.
+					// Not sure whether the check is significant in the first place.
```

Comment:
Do you mean that if SQ runs, we have enough confidence that the file was supplied?

Reply:
Yes.

Reply:
I originally added the tests because I thought it could be possible that SQ runs without errors, while not having any config supplied. How would we account for that? Disclaimer: I did not actually test if it would fail without config ...